external-crates/move/crates/move-analyzer/src/analysis/typing_analysis.rs

```diff
@@ -0,0 +1,942 @@
+        let name_start = self.lsp_start_position(&loc);
+        let const_info = self
+            .def_info
+            .get(&DefLoc::new(loc.file_hash(), name_start))
+            .unwrap();
+        let ident_type_def_loc = def_info_to_type_def_loc(self.mod_outer_defs, const_info);
+        self.use_defs.insert(
+            name_start.line,
+            UseDef::new(
+                self.references,
+                self.alias_lengths,
+                loc.file_hash(),
+                name_start,
+                loc.file_hash(),
+                name_start,
+                &constant_name.value(),
+                ident_type_def_loc,
+            ),
+        );
+        // scope must be passed here but it's not expected to be populated
+        self.visit_exp(&mut cdef.value);
+    }
+
+    fn visit_function(
+        &mut self,
+        _module: move_compiler::expansion::ast::ModuleIdent,
+        function_name: P::FunctionName,
+        fdef: &mut T::Function,
+    ) {
+        self.reset_for_module_member();
+        if ignored_function(function_name.value()) {
+            return;
+        }
+        let loc = function_name.loc();
+        // first, enter self-definition for function name (unwrap safe - done when inserting def)
+        let name_start = to_lsp_position(self.start_position(&loc));
+        let fun_info = self
+            .def_info
+            .get(&DefLoc::new(loc.file_hash(), name_start))
+            .unwrap();
+        let fun_type_def = def_info_to_type_def_loc(self.mod_outer_defs, fun_info);
+        let use_def = UseDef::new(
+            self.references,
+            self.alias_lengths,
+            loc.file_hash(),
+            name_start,
+            loc.file_hash(),
+            name_start,
+            &function_name.value(),
+            fun_type_def,
+        );
+
+        self.use_defs.insert(name_start.line, use_def);
+
+        // Get symbols for a function definition
+        for tp in &fdef.signature.type_parameters {
+            self.add_type_param(tp);
+        }
+
+        for (mutability, pname, ptype) in &mut fdef.signature.parameters {
+            self.visit_type(ptype);
+
+            // add definition of the parameter
+            self.add_local_def(
+                &pname.loc,
+                &pname.value.name,
+                ptype.clone(),
+                false, /* with_let */
+                matches!(mutability, E::Mutability::Mut(_)),
+            );
+        }
+
+        match &mut fdef.body.value {
+            T::FunctionBody_::Defined(seq) => {
+                self.visit_seq(seq);
+            }
+            T::FunctionBody_::Macro | T::FunctionBody_::Native => (),
+        }
+        // process return types
+        self.visit_type(&mut fdef.signature.return_type);
+
+        // clear type params from the scope
+        self.type_params.clear();
+    }
+
+    fn visit_lvalue_list(&mut self, kind: &LValueKind, lvalues: &mut T::LValueList) {
```

Comment:
Isn't it handled by the default visitor implementation?

Reply:
Yes. I had my IDE inline all the functions and deleted all the ones I didn't want to change the behavior of, but missed this one.